base/range.jl

```diff
@@ -828,17 +828,7 @@ function getindex(r::AbstractUnitRange, s::AbstractUnitRange{T}) where {T<:Integ
     @boundscheck checkbounds(r, s)
 
     if T === Bool
-        if length(s) == 0
-            return r
-        elseif length(s) == 1
-            if first(s)
-                return r
-            else
-                return range(r[1], length=0)
-            end
-        else # length(s) == 2
-            return range(r[2], length=1)
-        end
+        range(ifelse(first(s), first(r), last(r)), length = Int(last(s)))
```

Comment:
ah - interesting. Actually I thought it was the opposite as `ifelse` avoids branching - isn't it the point of `ifelse` existence? But maybe it is apparent only in tight loops. I will change.

Reply:
Branches are typically fast. Though I vaguely recall seeing mention that they were not as good on the Pentium 3? I don't have one though to check. But the syntax here bears only limited relevance to the way this will execute.